src/file_output.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{Sample, Source};
+use hound::{SampleFormat, WavSpec};
+use std::path;
+
+/// This procedure saves Source's output into a wav file. The output samples format is 32-bit float.
+/// This function is intended primarily for testing and diagnostics. It can be used to see
+/// the output without opening output stream to a real audio device.
+pub fn output_to_wav<S: Sample, P: AsRef<path::Path>, Src: Source<Item = S>>(
+    source: &mut Src,
+    wav_file: &P,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let format = WavSpec {
+        channels: source.channels(),
+        sample_rate: source.sample_rate(),
+        bits_per_sample: 32,
+        sample_format: SampleFormat::Float,
+    };
+    let mut writer = hound::WavWriter::create(wav_file, format)?;
+    for sample in source {
+        writer.write_sample(sample.to_f32())?;
+    }
+    writer.finalize()?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use super::output_to_wav;
+    use crate::Source;
+    use std::time::Duration;
+
+    #[test]
+    fn test_output_to_wav() {
```

Comment:
cool integration test could be: 
- generate some source in rodio (sine). 
- write it out to wav
- read it in 
- write the just read source out to another file
- check if they are identical (given wav is lossless they should be)

Reply:
> Also to make this transparent this would require to map cpal sample formats to hound parameters. It seems that this is not worth it until there are real use cases that need this.

ahh yeah I understand what you meant here now. Lets not then. Lets keep this simple :+1: